var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -130,10 +147,6 @@ def install_cpanm(self):
                 make()
                 make('install')
 
-    def setup_environment(self, spack_env, run_env):
-        """Set PERL5LIB to support activation of Perl packages"""
-        run_env.set('PERL5LIB', join_path(self.prefix, 'lib', 'perl5'))
```

Comment:
I don't *think* so.

Perl-extending packages that are not activated need to add their prefix to PERL5LIB, e.g. in their modulefile, if people want to use them.  That task lives in those packages, not the Perl package itself.

This was here because we were linking the activated packages into a directory that's wasn't normally part of `@INC`.  This fix arranges for our directory to be on `@INC` so we no longer depend on the environment variable (less opportunity for foot shooting).